mock/py/mockbuild/buildroot.py

```diff
@@ -40,6 +40,18 @@ def wrapper(self, *args, **kwargs):
     return wrapper
 
 
+def call_just_once(f):
+    """
+    Make sure that an object method is called just once.
+    """
+    def wrapper(self, *args, **kwargs):
+        if hasattr(self, "called_method_retval"):
+            return self.called_method_retval
+        self.called_method_retval = f(self, *args, **kwargs)
```

Comment:
This is a bit odd; you're saving the return value on `self`, which means that there is only ever one return value saved regardless of how many functions are decorated. It's also given a public naming convention even though it's not public.

Reply:
Also odd is that this passes through `*args, **kwargs`, but that is not used to differentiate return values.

Reply:
Good point. This would be very tricky in the future, thank you for the review! TIL